src/pages/api/search/[query].ts

```diff
@@ -0,0 +1,170 @@
+import { status as Status } from '@prisma/client';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+import type { Bounties } from '@/interface/listings';
+import { prisma } from '@/prisma';
+
+// GOTTA SAVE FROM SQL INJECTION
+function checkInvalidItems(obj: object, arr: string[]): boolean {
+  return arr.some((s) => !(s in obj));
+}
+
+function duplicateElements(array: string[], count: number) {
+  return array.flatMap((item) => Array(count).fill(item));
+}
+
+const Skills = {
+  DEVELOPMENT: 'DEVELOPMENT',
+  DESIGN: 'DESIGN',
+  CONTENT: 'CONTENT',
+  OTHER: 'OTHER',
+};
+
+export default async function user(req: NextApiRequest, res: NextApiResponse) {
+  const params = req.query;
+  const query = params.query as string;
```

Comment:
sanitize query using something like `query.replace(/[^a-zA-Z0-9_]/g, "")` 
helps prevent sql injection attacks (since we have written raw sql code)

Reply:
okay, have added validation on filters,
forgot on query.

injection wont happen either ways, coz using sql variables. 

but will still do the above